test/interface.jl

```diff
@@ -175,5 +175,61 @@ end
                 @test isa(err, Mooncake.ValueAndPullbackReturnTypeError)
             end
         end
+
+        # Test for new copy function
+        @testset for i in eachindex(additional_test_set)
```

Comment:
rather than having a comment, it might be better to include a string in the testset. Something like
```suggestion
        @testset "_copy_temp, $(typeof(i))" for i in eachindex(additional_test_set)
```

Reply:
sorry, using `eachindex` is almost a habit for me now lol.